modules/services/colima.nix

```diff
@@ -0,0 +1,267 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.colima;
+  yamlFormat = pkgs.formats.yaml { };
+in
+{
+  meta.maintainers = [
+    lib.hm.maintainers.will-lol
+  ];
+
+  options.services.colima = {
+    enable = lib.mkEnableOption "Colima, a container runtime";
+
+    package = lib.mkPackageOption pkgs "colima" { };
+    dockerPackage = lib.mkPackageOption pkgs "docker" {
+      extraDescription = "Used by colima to activate profiles. Not needed if no profile is set to isActive.";
+    };
+    perlPackage = lib.mkPackageOption pkgs "perl" {
+      extraDescription = "Used by colima during image download for the shasum command.";
+    };
+    sshPackage = lib.mkPackageOption pkgs "openssh" {
+      extraDescription = "Used by colima to manage the vm.";
+    };
+    coreutilsPackage = lib.mkPackageOption pkgs "coreutils" {
+      extraDescription = "Used in various ways by colima.";
+    };
+    curlPackage = lib.mkPackageOption pkgs "curl" {
+      extraDescription = "Used by colima to donwload images.";
+    };
+    bashPackage = lib.mkPackageOption pkgs "bashNonInteractive" {
+      extraDescription = "Used by colima's internal scripts.";
+    };
```

Comment:
Sorry, I got distracted... this just reminded me of the thing I wanted to tackle previously with a centralized dependencies management system so that we don't have a bunch of different modules doing this their own way... but, I won't block on something that doesn't yet exist. 

Another thing I was thinking,  is that maybe we could use the existing module's package definitions for things like docker, ssh, and bash ? But, might be weird having some just using values from another module's config.